tests/utils/spark_operator.go

```diff
@@ -1,135 +1,158 @@
 package utils
 
 import (
+	"errors"
+	"fmt"
 	log "github.com/sirupsen/logrus"
-	coreV1 "k8s.io/api/core/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"os"
 	"os/exec"
+	"path"
+	"regexp"
 	"strings"
+	"time"
 )
 
 type SparkOperatorInstallation struct {
+	Namespace    string
+	InstanceName string
+	Clients      *kubernetes.Clientset
+}
+
+type SparkInstance struct {
+	Name      string
 	Namespace string
-	Clients   *kubernetes.Clientset
+	Status    string
 }
 
-func InstallSparkOperator() *SparkOperatorInstallation {
+func InstallSparkOperator() (*SparkOperatorInstallation, error) {
 	return InstallSparkOperatorWithNamespace(DefaultNamespace)
 }
 
-func InstallSparkOperatorWithNamespace(namespace string) *SparkOperatorInstallation {
+func InstallSparkOperatorWithNamespace(namespace string) (*SparkOperatorInstallation, error) {
 	clientSet, err := GetK8sClientSet()
 	if err != nil {
 		log.Fatal(err)
 		panic(err)
 	}
 
-	err = installSparkOperatorWithHelm(namespace)
-	if err != nil {
-		log.Fatal(err)
-		panic(err)
-	}
-
 	spark := SparkOperatorInstallation{
 		Namespace: namespace,
 		Clients:   clientSet,
 	}
-	return &spark
-}
+	spark.CleanUp()
 
-func (spark *SparkOperatorInstallation) CleanUp() {
-	uninstallSparkOperatorWithHelm(spark.Namespace)
+	installScript := exec.Command("bash", path.Join(TestDir, "../scripts/install_operator.sh"))
+	if strings.Contains(OperatorImage, ":") {
+		operatorImage := strings.Split(OperatorImage, ":")
+		installScript.Env = append(os.Environ(),
+			"NAMESPACE="+namespace,
+			"OPERATOR_IMAGE_NAME="+operatorImage[0],
+			"OPERATOR_VERSION="+operatorImage[1])
+	} else {
+		installScript.Env = append(os.Environ(),
+			"NAMESPACE="+namespace,
+			"OPERATOR_IMAGE_NAME="+OperatorImage)
+	}
+	out, err := installScript.CombinedOutput()
+	log.Infof("KUDO Spark operation installation script output:\n%s", out)
+	if err != nil {
+		return nil, err
+	}
+
+	instanceNameRE, _ := regexp.Compile(`instance\.kudo\.dev/v1alpha1/(spark-\w{6}) created`)
+	parsedInstanceName := instanceNameRE.FindStringSubmatch(string(out))
+	spark.InstanceName = parsedInstanceName[1]
+
+	spark.waitForInstanceStatus("COMPLETE")
+
+	return &spark, nil
 }
 
-func (spark *SparkOperatorInstallation) OperatorPod() (coreV1.Pod, error) {
-	pods, err := spark.Clients.CoreV1().Pods(spark.Namespace).List(v1.ListOptions{LabelSelector: "app.kubernetes.io/name=sparkoperator"})
+func (spark *SparkOperatorInstallation) CleanUp() {
+	// So far multiple Spark operator instances in one namespace is not a supported configuration, so whole namespace will be cleaned
+	log.Infof("Uninstalling _all_ kudo spark operator instances and versions from %s", spark.Namespace)
```

Comment:
```suggestion
	log.Infof("Uninstalling ALL kudo spark operator instances and versions from %s", spark.Namespace)
```
Or does `_x_` actually get printed in italics?